### PR TITLE
Remove identifier types

### PR DESCRIPTION
**Pull request by iainmcgin**

### CODE REVIEW

File: demoproviders/trapdoor/java/org/openyolo/demoprovider/trapdoor/CredentialQueryReceiver.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ protected void processCredentialRequest(
         final QueryResponseSender responseSender = new QueryResponseSender(applicationContext);
 
         // Ensure the caller supports the Id and Password authentication method
```
Comment (dxslly) [inline demoproviders/trapdoor/java/org/openyolo/demoprovider/trapdoor/CredentialQueryReceiver.java]: Comment needs to be updated

File: testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java [hunk 2]
```diff
@@ -103,7 +103,7 @@ private void initialize(Context context) {
         }
 
         if (mIdAndPasswordCheckBox.isChecked()) {
```
Comment (dxslly) [inline testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java]: Update variable name

File: testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java [hunk 3]
```diff
@@ -103,7 +103,7 @@ private void initialize(Context context) {
...
-            authenticationMethods.add(AuthenticationMethods.ID_AND_PASSWORD);
+            authenticationMethods.add(AuthenticationMethods.EMAIL);
         }
```
Comment (dxslly) [inline testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java]: Doesn't have to be in this cl, but we should also update this view to reflect the new well defined authentication methods.

However we should update the UI for at least this checkbox field from "openyolo://id-and-password" to "openyolo://email"

File: testapp/java/org/openyolo/testapp/CredentialView.java [hunk 4]
```diff
@@ -152,7 +152,7 @@ void generatePassword() {
 
     @OnClick(R.id.openyolo_id_and_password_provider_button)
     void setIdAndPasswordAuthenticationMethod() {
```
Comment (dxslly) [inline testapp/java/org/openyolo/testapp/CredentialView.java]: Variable and method names need to be updated

File: demoproviders/barbican/java/org/openyolo/demoprovider/barbican/CredentialQualityScore.java [hunk 5]
```diff
@@ -64,12 +64,13 @@ private static int getQualityScore(
...
 
         // email identifiers are recognisable and useful for ID token acquisition
-        if (CredentialClassifier.isEmailIdentifier(id)) {
+        if (AuthenticationMethods.EMAIL.toString().equals(authMethod)) {
             score <<= 1;
         }
 
         // prefer credentials that don't use passwords
-        if (!AuthenticationMethods.ID_AND_PASSWORD.equals(authMethod)) {
+        if (!AuthenticationMethods.PHONE.toString().equals(authMethod)
```
Comment (dxslly) [inline demoproviders/barbican/java/org/openyolo/demoprovider/barbican/CredentialQualityScore.java]: Nit: Phone may/may not use passwords

File: spi/javatests/java/org/openyolo/spi/RetrieveIntentResultUtilTest.java [hunk 6]
```diff
@@ -38,17 +38,21 @@
...
+                EMAIL_ID,
+                EMAIL,
+                new AuthenticationDomain("https://www.example.com"))
+                .build();
+        Intent response = RetrieveIntentResultUtil.createResponseData(cr.getProto());
+
+        assertThat(response.hasExtra(ProtocolConstants.EXTRA_CREDENTIAL)).isTrue();
+        byte[] bytes = response.getExtras().getByteArray(ProtocolConstants.EXTRA_CREDENTIAL);
+        Credential credential = Credential.fromProtoBytes(bytes);
+        assertThat(credential.getIdentifier()).isEqualTo(cr.getIdentifier());
```
Comment (dxslly) [inline spi/javatests/java/org/openyolo/spi/RetrieveIntentResultUtilTest.java]: Please use the constant EMAIL_ID instead

File: demoproviders/trapdoor/java/org/openyolo/demoprovider/trapdoor/CredentialQueryReceiver.java (hunk 1 above)
Comment (iainmcgin) [inline demoproviders/trapdoor/java/org/openyolo/demoprovider/trapdoor/CredentialQueryReceiver.java]: The trapdoor provider generally needs an overhaul - it's a hint provider, but was written before we added the hint method. Anyway, fixed the comment and related issues in the meantime until I have time to do more invasive surgery.

File: testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java (hunk 2 above)
Comment (iainmcgin) [inline testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java]: Done

File: testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java (hunk 3 above)
Comment (iainmcgin) [inline testapp/java/org/openyolo/testapp/AuthenticationMethodsInputView.java]: For now, updated to just have email. Will open an issue to add phone and username.

File: testapp/java/org/openyolo/testapp/CredentialView.java (hunk 4 above)
Comment (iainmcgin) [inline testapp/java/org/openyolo/testapp/CredentialView.java]: Done

File: demoproviders/barbican/java/org/openyolo/demoprovider/barbican/CredentialQualityScore.java (hunk 5 above)
Comment (iainmcgin) [inline demoproviders/barbican/java/org/openyolo/demoprovider/barbican/CredentialQualityScore.java]: I'll update the comment - what I'm really saying here is "prefer federated auth methods".

File: spi/javatests/java/org/openyolo/spi/RetrieveIntentResultUtilTest.java (hunk 6 above)
Comment (iainmcgin) [inline spi/javatests/java/org/openyolo/spi/RetrieveIntentResultUtilTest.java]: Done

**Pull request merged**